destination/device_finder.go

```diff
@@ -83,3 +85,12 @@ func (d deviceFinder) FindDevice(destination Simulator) (Device, error) {
 
 	return device, err
 }
+
+// XcodebuildDestination returns the required xcodebuild -destination flag value for a device
+func (d Device) XcodebuildDestination() string {
+	if d.Arch == "" {
+		return fmt.Sprintf("id=%s", d.ID)
+	}
+
+	return fmt.Sprintf("id=%s,arch=%s", d.ID, d.Arch)
```

Comment:
Does this work? I thought we can't use the `id` at all with `arch` and need to use the rest of the arguments (`platform`, `os`, `device`).